core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -782,7 +782,7 @@ class LogTest {
     }
 
     // Retain snapshots for the last 2 segments
-    ProducerStateManager.deleteSnapshotsBefore(logDir, segmentOffsets(segmentOffsets.size - 2))
+    ProducerStateManager.listSnapshotFiles(logDir).filter(_.offset < segmentOffsets(segmentOffsets.size - 2)).foreach(_.deleteIfExists())
```

Comment:
Since deleteSnapshotsBefore() still exist, could we keep using it?

Reply:
Yes, it should work if we switch these back to using deleteSnapshotsBefore. Thanks!